src/internal/modules/charge-information/controllers/charge-information.js

```diff
@@ -286,6 +286,28 @@ const redirectToStartOfElementFlow = (request, h) => {
   return h.redirect(routing.getChargeElementStep(licenceId, chargeElementId, CHARGE_ELEMENT_FIRST_STEP, request.query));
 };
 
+const redirectToStartOfCategoryFlow = (request, h) => {
+  const { chargeVersionWorkflowId } = request.query;
+  const { licenceId } = request.params;
+  const { draftChargeInformation: currentState } = request.pre;
+
+  // Create new element to edit in the session state
+  const id = uuid();
+  const data = currentState.chargeElements.reduce((acc, element) => {
+    element.scheme === 'alcs'
```

Comment:
I'm not 100% sure what's happening here.

if the element has a scheme value of `alcs`, why are we doing `acc.chargePurposes.push({ ...element, scheme: 'sroc' })`?

Is this a post-SROC only thing? If so, is it worth adding a condition here to check the isSrocLive flag?

Reply:
Yes this is a SROC controller method. The access to this method is controlled via the SROC feature flag on the view.